app/controllers/posts_controller.rb

```diff
@@ -1,8 +1,8 @@
 class PostsController < ApplicationController
   def show
-    @post = Post.find(params[:id])
-    @user = @post.author
-    @comments = @post.comments.includes(:author)
+    @post = Post.includes(:author).find(params[:id])
+    @user = User.find(@post.author_id)
+    @comments = Comment.includes(:author).where(post_id: @post.id)
   end
 
   def index
```

Comment:
- [ ] in the post index action you should use `@posts = @user.posts.includes(:comments)` to fix the n+1 issue

Reply:
@mikethreels  there currently no n+1 problems your solution actually causes one when I use it

![Screenshot from 2022-12-08 00-18-51](https://user-images.githubusercontent.com/86778388/206309748-3dc2f0e7-233e-4efa-8bf9-87286408a93b.png)